### ISSUE TITLE
2909: RuntimeWarning: Mean of empty slice.

### ISSUE THREAD

**Issue by togtog708908** [facebookresearch/MUSE#54]
I tried unsupervised.py by wiki.en.vec  and wiki.ja.vec pretrained from fasttext.
`python unsupervised.py --src_lang en --tgt_lang ja --src_emb wiki.en.vec  --tgt_emb wiki.ja.vec --n_refinement 5  --cuda 1 --exp_path vec --dico_eval en-ja.5000-6500.txt --normalize_embeddings center`
but some Warning and k=0
`INFO - 06/06/18 17:59:04 - 0:10:32 - 996000 - Discriminator loss: 0.3396 - 3339 samples/s
INFO - 06/06/18 17:59:07 - 0:10:35 - ====================================================================
INFO - 06/06/18 17:59:07 - 0:10:35 -                        Dataset      Found     Not found          Rho
INFO - 06/06/18 17:59:07 - 0:10:35 - ====================================================================
INFO - 06/06/18 17:59:07 - 0:10:35 - ====================================================================
/***.pyenv/versions/3.6.1/lib/python3.6/site-packages/numpy/core/fromnumeric.py:2909: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/***/.pyenv/versions/3.6.1/lib/python3.6/site-packages/numpy/core/_methods.py:80: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
INFO - 06/06/18 17:59:07 - 0:10:35 - Monolingual source word similarity score average: nan
INFO - 06/06/18 17:59:07 - 0:10:35 - Found 1799 pairs of words in the dictionary (1459 unique). 0 other pairs contained at least one unknown word (0 in lang1, 0 in lang2)
INFO - 06/06/18 17:59:07 - 0:10:35 - 1459 source words - nn - Precision at k = 1: 0.000000
INFO - 06/06/18 17:59:08 - 0:10:35 - 1459 source words - nn - Precision at k = 5: 0.000000
INFO - 06/06/18 17:59:08 - 0:10:35 - 1459 source words - nn - Precision at k = 10: 0.000000
INFO - 06/06/18 17:59:08 - 0:10:35 - Found 1799 pairs of words in the dictionary (1459 unique). 0 other pairs contained at least one unknown word (0 in lang1, 0 in lang2)
INFO - 06/06/18 17:59:38 - 0:11:06 - 1459 source words - csls_knn_10 - Precision at k = 1: 0.000000
INFO - 06/06/18 17:59:38 - 0:11:06 - 1459 source words - csls_knn_10 - Precision at k = 5: 0.068540
INFO - 06/06/18 17:59:39 - 0:11:06 - 1459 source words - csls_knn_10 - Precision at k = 10: 0.068540`

**Comment by glample** [comment]
Hi,

Can you try to add `--dico_build S2T` to your command? The default setting is `S2T&T2S` which takes the intersection of source-to-target and target-to-source translations. But if the alignment is bad, the intersection might be empty and this will create an issue.

If not, can you send me your embeddings? I'll have a look.

**Comment by glample** [comment]
This is running fine with me. What versions of Python / PyTorch are you using?
Can you send me the entire train.log?

**Comment by togtog708908** [comment]
I send you my train.log

**Comment by glample** [comment]
From what I see in the log there is no problem. The "mean of empty slice" seems to come from the word similarity evaluation, because there is no word similarity dataset available in Japanese. If you disable this evaluation I assume you won't see any warning. Regarding the translation accuracy, I realized that the accuracy is already pretty bad even in the supervised setting, so I guess that the model is just not able to do the alignment. Maybe you can try more in-domain embeddings if you have some, or the last fastText embeddings that are trained on Common Crawl and are of better quality than the Wikipedia ones.

**Comment by togtog708908** [comment]
Thanks.
I tried it on another machine. It is a very bad result, but it worked.

**Comment by glample** [comment]
What accuracy do you get?
Do you get good results for more similar language pairs (like en-es)?

**Comment by ccclyu** [comment]
Hi Guillaume, I have met with the same issue but the difference is that I use fasttext commoncrawl pretrained [word embedding](https://fasttext.cc/docs/en/crawl-vectors.html) to align english and Japanese.  Any suggestions?
